### PR TITLE
Bump Kani version to 0.29.0

### PR DESCRIPTION
**Pull request by tautschnig**
### Description of changes: 

Mark current revision as release 0.29.0.

### Resolved issues:

n/a

### Related RFC:

n/a

### Call-outs:

n/a

### Testing:

* How is this change tested? CI

* Is this a refactor change? no

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- n/a Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,24 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.29.0]
+
+### What's Changed
+* Fix symtab json file removal and reduce regression scope ([pull request](https://github.com/model-checking/kani/pull/2447) by @celinval)
+* Fix regression on concrete playback inplace ([pull request](https://github.com/model-checking/kani/pull/2454) by @celinval)
+* Update rust toolchain to 2023-04-29 ([pull request](https://github.com/model-checking/kani/pull/2452) by @zhassan-aws)
+* Update stubbing.md with the right command ([pull request](https://github.com/model-checking/kani/pull/2460) by @jaisnan)
```
Comment (celinval) [inline CHANGELOG.md:13]: I believe we skip documentation updates

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,24 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.29.0]
+
+### What's Changed
+* Fix symtab json file removal and reduce regression scope ([pull request](https://github.com/model-checking/kani/pull/2447) by @celinval)
+* Fix regression on concrete playback inplace ([pull request](https://github.com/model-checking/kani/pull/2454) by @celinval)
+* Update rust toolchain to 2023-04-29 ([pull request](https://github.com/model-checking/kani/pull/2452) by @zhassan-aws)
```
Comment (celinval) [inline CHANGELOG.md:12]: We put dependencies update at the bottom of the lis

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,24 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
+* Update stubbing.md with the right command ([pull request](https://github.com/model-checking/kani/pull/2460) by @jaisnan)
+* Create a playback command to make it easier to run Kani generated tests ([pull request](https://github.com/model-checking/kani/pull/2464) by @celinval)
+* Fix static variable initialization when they have the same value ([pull request](https://github.com/model-checking/kani/pull/2469) by @celinval)
+* Add jq to deps ([pull request](https://github.com/model-checking/kani/pull/2471) by @zhassan-aws)
```
Comment (celinval) [inline CHANGELOG.md:16]: We don't include infrastructure and internal changes

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,24 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
+* Update concrete playback documentation ([pull request](https://github.com/model-checking/kani/pull/2465) by @celinval)
```
Comment (celinval) [inline CHANGELOG.md:17]: Skip doc changes

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,24 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.29.0]
+
+### What's Changed
+* Fix symtab json file removal and reduce regression scope ([pull request](https://github.com/model-checking/kani/pull/2447) by @celinval)
+* Fix regression on concrete playback inplace ([pull request](https://github.com/model-checking/kani/pull/2454) by @celinval)
+* Update rust toolchain to 2023-04-29 ([pull request](https://github.com/model-checking/kani/pull/2452) by @zhassan-aws)
+* Update stubbing.md with the right command ([pull request](https://github.com/model-checking/kani/pull/2460) by @jaisnan)
+* Create a playback command to make it easier to run Kani generated tests ([pull request](https://github.com/model-checking/kani/pull/2464) by @celinval)
```
Comment (celinval) [inline CHANGELOG.md:14]: I wonder if we should add this as a major change. I don't think we have a guideline regarding new subcommands.

**Comment by tautschnig** [comment]
The following perf change arises with just the version number change, making the perf CI step fail:

<h2>solver_runtime</h2>

Benchmark | kani_new | kani_old | % change old → new
-- | -- | -- | --
s2n-quic/quic/s2n-quic-core/random::tests::gen_range_biased_test | 12.8224221 | 4.612618100000001 | +177.986%

**Comment by karkhaz** [comment]
> The following perf change arises with just the version number change, making the perf CI step fail

Thanks, I suggest that we just merge it, unless you also saw this benchmark regress when you previously force-pushed the commits

**Review by karkhaz — APPROVED** [review]

**Comment by zhassan-aws** [comment]
I've seen that on previous PRs as well. The performance of this test is flaky.

**Review by celinval — APPROVED** [review]
You should be able to merge this. I don't think you need to add the fixes to the playback sub-command since the bugs never hit a release.

**Review by feliperodri — APPROVED** [review]

**Pull request merged**